pillarbox-core-business/src/main/java/ch/srgssr/pillarbox/core/business/MediaCompositionMediaItemSource.kt

```diff
@@ -151,6 +152,43 @@ class MediaCompositionMediaItemSource(
         }
     }
 
+    /**
+     * Service used to get a scaled image URL. This only works for SRG images.
+     *
+     * @param baseUrl Base URL of the service.
+     */
+    internal class ImageScalingService(
+        private val baseUrl: URL = IlHost.DEFAULT
+    ) {
+        /**
+         * The supported widths.
+         *
+         * @property width The width in pixels.
+         */
+        enum class ImageWidth(@Px val width: Int) {
+            W240(width = 240),
+            W320(width = 320),
+            W480(width = 480),
+            W960(width = 960),
+            W1920(width = 1920)
+        }
+
+        fun getScaledImageUrl(
+            imageUrl: String,
+            width: ImageWidth,
+        ): String {
+            return URLBuilder(baseUrl.toString())
```

Comment:
Compared to the initial implementation of this method, we now use Ktor's `URLBuilder` instead of `Uri`, so we can test it in regular unit tests.

Reply:
Sounds good to me